spec/services/record_declaration_spec.rb

```diff
@@ -293,6 +293,36 @@
     end
   end
 
+  context "when re-posting after a clawback" do
+    let(:lead_provider) { cpd_lead_provider.lead_provider }
+    let(:participant_profile) { create(:ect, :eligible_for_funding, lead_provider:) }
+    let(:schedule) { Finance::Schedule::ECF.find_by(schedule_identifier: "ecf-standard-september") }
+    let(:declaration_date) { schedule.milestones.find_by(declaration_type: "started").start_date }
+    let(:course_identifier) { "ecf-induction" }
+
+    before do
+      create(:ecf_statement, :output_fee, deadline_date: 6.weeks.from_now, cpd_lead_provider:)
+    end
+
+    it "creates the second declaration" do
+      participant_declaration = described_class.new(params).call
+
+      statement = participant_declaration.statements[0]
+
+      service = ParticipantDeclarations::MarkAsPayable.new(statement)
+      service.call(participant_declaration)
+
+      service = ParticipantDeclarations::MarkAsPaid.new(statement)
+      service.call(participant_declaration)
+
+      Finance::ClawbackDeclaration.new(participant_declaration).call
```

Comment:
could we use the `clawed_back` trait for declarations instead of setting this up manually?

Reply:
@StupidCodeFactory do you have a preference here of going through the service classes or setting up a record manually?